scipy/stats/_stats_py.py

```diff
@@ -4066,8 +4079,6 @@ def f_oneway(*samples, axis=0):
         raise TypeError('at least two inputs are required;'
                         f' got {len(samples)}.')
 
-    samples = [np.asarray(sample, dtype=float) for sample in samples]
```

Comment:
Can you remind me why this is removed? Probably because the decorator ensures that everything is at least floating point, but if this is an effort to improve dtype handling, remember that we will eventually take another pass for just that.

And when we do, we would determine the result dtype here and convert everything as needed at the beginning. We would keep the in-place operations below, I think.

Reply:
> Probably because the decorator ensures that everything is at least floating point

Yes, and we have done something similar for other functions too. For example, see #19425, #19487. The only different here is that the function also ensures all the inputs are float64 before computing. I just thought that's an arbitrary requirement and removed this.